### DISCUSSION TITLE
How to restore using .frm and .ibd files in Laragon?

### DISCUSSION THREAD

**Discussion by MunbongChoi** [leokhoa/laragon#458] [Q&A]
Hello Laragon Community
I am currently a teaching assistant using Laragon to teach students.
This is my first time using Laragon but I'm having a bit of trouble.

I have .frm, .ibd files extracted from an existing database.
In the program I was using before, if I pasted the .frm, .ibd files into the new database directory, they were automatically recognized and used.
So I tried to recover them by pasting them into laragon/data like I did before, but the files were not recognized.

So I looked for another way and found that for .frm files, I could recover them using mysqlfrm, but mysqlfrm doesn't exist and doesn't work.

Is it by any chance impossible to recover using .ibd and .frm files in laragon?

If it is possible, I would be grateful if you could tell me how to do it. 
Thank you for reading.

**Comment by MunbongChoi** [comment]
I'm still waiting every answers....

**Reply by Pen-y-Fan** [reply]
Copying files isn't the usual way to backup and restore a MySQL database. According to https://fileinfo.com/extension/ibd#:~:text=An%20IBD%20file%20is%20a,open%2Dsource%20database%20management%20system. the files are **often**  encrypted. It does say MySQL can open these file types, but refer to the [documentation](https://dev.mysql.com/doc/).

The standard way to backup and restore from a MySQL database is using mysqldump. See these examples:

- https://simplebackups.com/blog/the-complete-mysqldump-guide-with-examples/

**Reply by MunbongChoi** [reply]
thanks to your reply. I was decided to create new databases. Now I must be sleep right now. thank you
